### PR TITLE
return the toString of invocation handler when calling toString on thee proxy

### PR DESCRIPTION
**Pull request by juddgaddie**
see #15

### CODE REVIEW

File: src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java [hunk 1]
```diff
@@ -51,6 +53,11 @@
     @Override
     public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable
     {
+        if (TO_STRING_METHOD_NAME.equals(method.getName()))
```
Comment (mikeb01) [inline src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java]: To be completely accurate, I think this check should also check the number of parameters to the call is zero.  An overload of Foo.toString(long l) may need to be proxied.

Comment (juddgaddie) [inline src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java]: Good point. I'll add that.

File: src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java [hunk 2]
```diff
@@ -53,7 +53,7 @@
     @Override
     public Object invoke(final Object proxy, final Method method, final Object[] args) throws Throwable
     {
-        if (TO_STRING_METHOD_NAME.equals(method.getName()))
+        if (TO_STRING_METHOD_NAME.equals(method.getName()) && args == null)
         {
```
Comment (mikeb01) [inline src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java]: I think that you need to check args.length == 0.  Could you include a unit test to verify?

Comment (juddgaddie) [inline src/main/java/com/lmax/tool/disruptor/reflect/ReflectiveRingBufferInvocationHandler.java]: It is definitely null when there are no arguments, I've tested it and the JavaDoc states: 
```
args - an array of objects containing the values of the arguments passed in the method invocation on the proxy instance, or null if interface method takes no arguments.
```

I'll work out a way of putting in a meaningful unit test to verify.

**Comment by juddgaddie** [comment]
@epickrram reviewed and is happy for me to merge.

**Pull request merged**